src/Mod/Sketcher/Gui/Utils.cpp

```diff
@@ -327,6 +327,48 @@ double SketcherGui::GetPointAngle(const Base::Vector2d& p1, const Base::Vector2d
     return dY >= 0 ? atan2(dY, dX) : atan2(dY, dX) + 2 * M_PI;
 }
 
+// Set the two points on circles at minimal distance
+// in concentric case return false and set points on relative X axis
+bool SketcherGui::GetCirclesMinimalDistance(const Part::GeomCircle *circle1, const Part::GeomCircle *circle2, Base::Vector3d *point1, Base::Vector3d *point2)
+{
+    auto center1 = circle1->getCenter();
+    auto center2 = circle2->getCenter();
+    double radius1 = circle1->getRadius();
+    double radius2 = circle2->getRadius();
+    Base::Vector3d v = center2 - center1;
+    double length = v.Length();
+    auto p1 = Base::Vector3d(0., 0., 0.);
+    auto p2 = Base::Vector3d(0., 0., 0.);
+    bool retval = false;
+
+    v = v.Normalize();
+    if (length == 0) { //concentric case
```

Comment:
Shall we manage `Confusion` here ?

Reply:
I would have said that length == 0 is always false, if it weren't because the UI paints that case. So the solver is that good in my simple case.

However, what would happen if length is not zero, but very close to zero. Well, v will be a normalized vector and it will be treated under the corresponding inner case. That looks like ok.